### ISSUE TITLE
shell syntax highlighting gets confused with single quoted string in bracket pattern

### ISSUE THREAD

**Issue by calestyo** [vim/vim#8244]
**Describe the bug**
Shell syntax highlighting is broken by a like like this:
```
if [ -n "${str##*['$`\']*}"  -o  -z "${str}" ]; then
```

it seems to think that all after the 2nd `'` would be a string, while in the above, one has a pattern with a bracket that contains a single quoted string.

**Environment (please complete the following information):**
 - Vim version 8.2.2434
 - OS: Debian Sid
 - Terminal: GNOME Terminal
 
Cheers,
Chris.

**Comment by cecamp** [comment]
Is that backquote in your expression intentional? 
Which shell is this for: ksh, bash, or bourne shell?
Should there be two backslashes?

**Comment by calestyo** [comment]
Yes it is intentional,... i.e. I match for any of `$`, ` or `\`.

This should be valid in at least bash and POSIX sh.

Two `\ `would not make sense, it's a single quoted $`\ in a bracket expression in a pattern, doubling it woud be as if you write `[a-za-z]`.

**Comment by calestyo** [comment]
(github doesn't get the ` markup right...)

**Comment by k-takata** [comment]
I edited your comment. Is this intended?

If you want to write a ``` ` ``` in the code style, you should put it in triple backquotes or in double backquotes: `` ``` ` ``` `` or ``` `` ` `` ```.

**Comment by calestyo** [comment]
Yes, thanks... now it should be right :-)

**Comment by zzzyxwvut** [comment]
It is taken care of as of `v9.1.0980~1` (#15941).
